### PR TITLE
build 0.19b2

### PR DESCRIPTION
**Pull request by amueller**
cc @ogrisel

### CODE REVIEW

**Comment by amueller** [comment]
this is copied from master basically. It looks like it's only building one numpy version. Is that the standard?

File: circle.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 checkout:
   post:
-    - ./ci_support/checkout_merge_commit.sh
+    - ./ci_support/fast_finish_ci_pr_build.sh
```
Comment (jakirkham) [inline circle.yml]: Weird that we are losing `checkout_merge_commit.sh`. Could you please try re-rendering again?

**Comment by jakirkham** [comment]
Not going to have lots of time to look at this near term as I'm buried with other work.

> It looks like it's only building one numpy version. Is that the standard?

If you recall, we had to create separate branches for each NumPy version because of the time limits with CircleCI. ( https://github.com/conda-forge/scikit-learn-feedstock/pull/52 ) So we will likely need to do the same thing with the `rc` branch.

**Comment by amueller** [comment]
oh, right...

**Comment by amueller** [comment]
rerendered

**Comment by jakirkham** [comment]
Looks better. Thanks.

Yeah, we should have a better solution for the NumPy matrix once PR ( https://github.com/conda-forge/conda-smithy/pull/532 ) is merged. Though there has been some discussion about doing NumPy pinning totally differently, which could also help if it works well. ( https://github.com/conda-forge/pandas-feedstock/pull/32 ) Could try the different pinning in this release as well if you would like.

**Comment by jakirkham** [comment]
Actually ignore the `pandas` suggestion as it doesn't work here due to `blas` support. We would need to go to multiple branches.

**Comment by amueller** [comment]
So merge this and then do multiple branches for the different numpy releases?

**Comment by jakirkham** [comment]
We can do that. While this is a bit unorthodox, I'm kind of want to re-rendering with the development version of `conda-smithy` to get CircleCI matrix support to see how that goes. Thoughts?

**Comment by amueller** [comment]
yeah sure, go for it.

**Comment by jakirkham** [comment]
Done. Unfortunately CircleCI puts each build in a separate status. That said, it is tolerable given we actually get a build matrix. :)

**Comment by jakirkham** [comment]
There seems to be a compilation error. Thoughts?

```
Error compiling Cython file:
------------------------------------------------------------
...

    cdef inline DTYPE_t dist(self, DTYPE_t* x1, DTYPE_t* x2,
                             ITYPE_t size) except -1 with gil:
        cdef np.ndarray x1arr
        cdef np.ndarray x2arr
        with gil:
            ^
------------------------------------------------------------
```

**Comment by ogrisel** [comment]
Indeed a change in cython 0.26 makes this construct invalid. This has been fixed in master and needs to be backported to the 0.19.X branch. Would it be possible to build 0.19b2 with Cython 0.25.2 in the mean time? It's just a build dependency, scikit-learn does not need cython at runtime.

**Comment by jakirkham** [comment]
Sorry fixed in which `master`? `scikit-learn` or `cython`?

In either case, pinning `cython` to an older version is perfectly acceptable.

**Comment by amueller** [comment]
Fixed in scikit-learn master.

**Comment by jakirkham** [comment]
Thanks for the info. Have re-rendered with development version of `conda-smithy` again to get the matrix back. I don't think there will be any need for further re-renderings, but we can discuss if needed.

**Comment by amueller** [comment]
Thanks :)

**Comment by ogrisel** [comment]
LGTM.

I backported the GIL fix in the scikit-learn 0.19.X branch so that 0.19 final won't have this problem anymore.

**Pull request closed without merging**